src/packed-pack.ts

```diff
@@ -283,19 +280,101 @@ export class PackContent {
                 reader.offset = relOffset;
             }
 
-            // n-byte type and length (3-bit type, (n-1)*7+4-bit length) compressed data
-            // https://git-scm.com/docs/pack-format#_size_encoding
-            const int = readVarIntLE(reader);
-            const btype = ((int >> 4) & 0b0111) as PackedType;
-            const length = ((int >> 7) << 4) | (int & 0b1111);
+            const objectPrelude = await this.readObjectPrelude(offsetBase, reader);
```

Comment:
`readObjectPrelude()` was deliberately not used here, because it too expensive (does more work than needed, and with your changes it also read delta objects) and performs unnecessary validations (we count objects here, not validate). So, for performance reasons (since we scan for all the objects, we need perform it as fast as possible) it's ok to copy 3 lines of code. This also good for memory consumption, since no any object allocations/disposals, no function invocations (call stack), no async/await are occur.
I think we need add a comment with notes to avoid changing this fragment in future.